src/states/adult/mod.rs

```diff
@@ -539,63 +459,127 @@ impl Base for Adult {
         Transition::Stay
     }
 
-    fn handle_direct_message(
+    fn handle_message(
         &mut self,
-        msg: Variant,
-        p2p_node: P2pNode,
+        sender: Option<ConnectionInfo>,
+        msg: Message,
         outbox: &mut dyn EventBox,
-    ) -> Result<Transition, RoutingError> {
-        use crate::messages::Variant::*;
-        match msg {
-            MessageSignature(msg) => self.handle_message_signature(*msg, *p2p_node.public_id()),
-            ParsecRequest(version, par_request) => {
-                self.handle_parsec_request(version, par_request, p2p_node, outbox)
+    ) -> Result<Transition> {
+        trace!("{} - Handle message {:?}", self, msg);
+
+        match msg.variant {
+            Variant::GenesisUpdate(info) => self.handle_genesis_update(*info),
```

Comment:
Should it check that the source was a section, and the destination itself?
i.e like for Elder messages we probably should check the type of source/destination is as expected for all messages.